src/Pages/GlobalConfigurations/Authorization/Shared/components/AppPermissions/types.ts

```diff
@@ -82,3 +89,39 @@ export interface AppOrJobSelectorProps
     > {
     setWorkflowList: Dispatch<SetStateAction<WorkflowListType>>
 }
+
+export interface RoleSelectorToggleConfig {
+    baseRole: boolean
+    accessManagerRoles: boolean
+}
+
+export interface RoleSelectorProps {
+    permission: DirectPermissionsRoleFilter
+    handleUpdateDirectPermissionRoleConfig: (updatedRoleConfig: UserRoleConfig) => void
+}
+export interface JobRoleSelectorProps {
```

Comment:
We should create some common type for these. If required we can extend these from that